drgn_tools/nfs_tools.py

```diff
@@ -40,14 +40,81 @@
 """ include/linux/sunrpc/sched.h """
 
 
-class RpcTaskState(BitNumberFlags):
+class RpcTaskState_1(BitNumberFlags):
+    RPC_TASK_RUNNING = 0
+    RPC_TASK_QUEUED = 1
+    RPC_TASK_ACTIVE = 2
+
+
+# 729749bb8da1 SUNRPC: Don't hold the transport lock across socket copy operations
+class RpcTaskState_2(BitNumberFlags):
     RPC_TASK_RUNNING = 0
     RPC_TASK_QUEUED = 1
     RPC_TASK_ACTIVE = 2
     RPC_TASK_MSG_RECV = 3
     RPC_TASK_MSG_RECV_WAIT = 4
 
 
+# 7ebbbc6e7bd0 SUNRPC: Simplify identification of when the message send/receive is complete
+class RpcTaskState_3(BitNumberFlags):
+    RPC_TASK_RUNNING = 0
+    RPC_TASK_QUEUED = 1
+    RPC_TASK_ACTIVE = 2
+    RPC_TASK_NEED_XMIT = 3
+    RPC_TASK_NEED_RECV = 4
+    RPC_TASK_MSG_RECV = 5
+    RPC_TASK_MSG_RECV_WAIT = 6
+
+
+# cf9946cd6144 SUNRPC: Refactor the transport request pinning
+class RpcTaskState_4(BitNumberFlags):
+    RPC_TASK_RUNNING = 0
+    RPC_TASK_QUEUED = 1
+    RPC_TASK_ACTIVE = 2
+    RPC_TASK_NEED_XMIT = 3
+    RPC_TASK_NEED_RECV = 4
+    RPC_TASK_MSG_PIN_WAIT = 5
+
+
+# ae67bd3821bb SUNRPC: Fix up task signalling
+class RpcTaskState_5(BitNumberFlags):
+    RPC_TASK_RUNNING = 0
+    RPC_TASK_QUEUED = 1
+    RPC_TASK_ACTIVE = 2
+    RPC_TASK_NEED_XMIT = 3
+    RPC_TASK_NEED_RECV = 4
+    RPC_TASK_MSG_PIN_WAIT = 5
+    RPC_TASK_SIGNALLED = 6
+
+
+def does_func_exist(prog: drgn.Program, func: str) -> bool:
+    try:
+        _ = prog[func]
+        return True
+    except KeyError:
+        return False
```

Comment:
This doesn't actually work for CTF, since CTF doesn't always contain the type definitions of functions. So looking up a function as a _drgn object_ doesn't reliably work with CTF. However, you can look up the corresponding _symbol_ which will be there for both CTF and DWARF.
```suggestion
    try:
        prog.symbol(func)
        return True
    except LookupError:
        return False
```

Reply:
Thanks Stephen, I made the suggested changes.

Reply:
I've retried it, I'm sure it was just a transient error, no worries!